src/pybind/mgr/dashboard/tox.ini

```diff
@@ -1,29 +1,60 @@
 [tox]
-envlist = py27-{cov,lint,run},py3-{cov,lint,run}
+envlist = py{27,36}-{cov,lint,run},clean,report
 skipsdist = true
-toxworkdir = {env:CEPH_BUILD_DIR}/dashboard
 minversion = 2.8.1
 
 [testenv]
+deps =
+    -rrequirements.txt
+    mock
+    pytest<4
+    pytest-cov
 setenv=
     CFLAGS = -DXMLSEC_NO_SIZE_T
     UNITTEST = true
     WEBTEST_INTERACTIVE = false
     LD_LIBRARY_PATH = {toxinidir}/../../../../build/lib
     PATH = {toxinidir}/../../../../build/bin:$PATH
-    py27: PYTHONPATH = {toxinidir}/../../../../build/lib/cython_modules/lib.2
-    py3:  PYTHONPATH = {toxinidir}/../../../../build/lib/cython_modules/lib.3
-    cov:  UNITTEST = true
-    cov:  COVERAGE_FILE = .coverage.{envname}
-commands=
-    pip install -r {toxinidir}/requirements.txt
-    py27: pip install -r {toxinidir}/requirements-py27.txt
-    py3: pip install -r {toxinidir}/requirements-py3.txt
-    cov: coverage erase
-    cov: {envbindir}/py.test --cov=. --cov-report= --junitxml=junit.{envname}.xml --doctest-modules controllers/rbd.py services/ tests/ tools.py
-    cov: coverage combine {toxinidir}/{env:COVERAGE_FILE}
-    cov: coverage report
-    cov: coverage xml
-    lint: pylint --rcfile=.pylintrc --jobs=5 . module.py tools.py controllers tests services exceptions.py grafana.py
-    lint: pycodestyle --max-line-length=100 --exclude=.tox,venv,frontend,.vscode --ignore=E402,E121,E123,E126,E226,E24,E704,W503,E741 .
-    run: {posargs}
+commands =
+    cov: pytest --cov --cov-append --cov-report=term-missing --doctest-modules --ignore=module.py --ignore=frontend/ {posargs}
```

Comment:
Can we get rid of the "Missing" column in coverage report?
IMHO is a little bit overwhelming:
```
  Name                                   Stmts   Miss  Cover   Missing
--------------------------------------------------------------------
awsauth.py                                80     57    29%   75-84, 87-96, 99-183
conftest.py                               12      0   100%
controllers/__init__.py                  545     79    86%   31, 34, 37, 40, 54, 77, 80, 137-139, 151, 154, 197, 204, 220, 251-252, 266, 280-282, 306, 308, 336-354, 406, 435-438, 449, 483-486, 564, 575, 583-596, 653-654, 663, 667, 671, 675-677, 681-684, 688, 773, 785, 796, 811, 814, 867, 891-894, 899-900
controllers/auth.py                       37     21    43%   19, 21-33, 40-45, 50, 52, 57-63
controllers/cephfs.py                    140    117    16%   19-23, 26-27, 30-32, 36-38, 50-77, 81-84, 89-101, 105-234, 247-279, 287-288, 292
controllers/cluster_configuration.py      66     45    32%   22-32, 35-36, 39, 43-55, 59-66, 69-72, 77, 80-83, 87-91, 94-102
controllers/docs.py                      217     50    77%   55, 69, 71, 73, 75, 90-97, 124-135, 157-160, 162, 312-329, 369-390, 451
controllers/erasure_code_profile.py       38      8    79%   41-42, 45-46, 50, 56-58
controllers/grafana.py                    27      2    93%   29, 43
controllers/health.py                    121     73    40%   25, 29, 36-64, 68-79, 83-90, 95-103, 106, 109-118, 121, 124, 129, 131-132, 136, 138-143, 147-161, 167-169, 173, 182, 188
controllers/host.py                        8      1    88%   12
controllers/iscsi.py                     474     32    93%   33-43, 46, 51-59, 62, 67, 73, 91, 112, 119-120, 136, 139, 348, 652
controllers/logging.py                     7      1    86%   10
controllers/logs.py                       32     17    47%   18-21, 24-27, 30-33, 36-42, 47-48
controllers/mgr_modules.py                77     55    29%   22-36, 46-52, 63-67, 77-78, 89-90, 102-103, 114-120, 131-132, 143-144, 150-168
controllers/monitor.py                    24     15    38%   16-36
controllers/nfsganesha.py                123     70    43%   98-105, 116-121, 131-153, 162-166, 177-209, 223-231, 248-266, 273, 277, 281-297, 301, 305, 309
controllers/osd.py                       107     43    60%   23, 29, 35, 43, 57-64, 74-78, 92, 101, 105, 122, 133, 145, 158, 170, 178-182, 187-188, 199-206, 209, 217-229
controllers/perf_counters.py              51     18    65%   16-38, 85
controllers/pool.py                      115     83    28%   24-44, 47-55, 58, 62-66, 70-72, 77, 82-83, 89-94, 97-131, 134-141, 146, 154-182
controllers/prometheus.py                 34      0   100%
controllers/rbd.py                       307    169    45%   37, 42-45, 59-61, 64, 81-86, 89, 92, 98-101, 105-116, 121, 126-127, 130, 137-151, 158-182, 187-198, 201-213, 218, 223-227, 234, 240-250, 256-257, 261-265, 275-280, 285-294, 307-310, 314-326, 333-336, 340-341, 350, 356, 367-370, 379, 387-389, 395-397, 404-406, 423-427, 437-442, 452-455, 463, 466-473, 479, 488-492, 501, 509
controllers/rbd_mirroring.py             254     85    67%   54, 124, 128, 132, 153-167, 171, 174, 251, 254-255, 279, 288-307, 323-324, 351, 353-356, 360, 364-372, 382, 384,
[...]
```

Reply:
Removed. I liked it because it allows you to easily work on improving coverage without the need of using the coverage.xml/html reports.